### PR TITLE
Disable optimizations based on lifetime intrinsics

### PR DESCRIPTION
**Pull request by aschwaighofer**
Lifetime sinking is broken because it ignores initializations via back-edges.
And optimizations baased on the lifetime intrinsic is broken because it ignores certain escapes.
rdar://83411917

### CODE REVIEW

File: llvm/lib/Transforms/Coroutines/CoroFrame.cpp [hunk 1]
```diff
@@ -2369,7 +2369,7 @@ void coro::buildCoroutineFrame(Function &F, Shape &Shape) {
     }
   }
 
-  sinkLifetimeStartMarkers(F, Shape, Checker);
+  //sinkLifetimeStartMarkers(F, Shape, Checker);
```
Comment (fhahn) [inline llvm/lib/Transforms/Coroutines/CoroFrame.cpp]: Would it be possible to add a unit test for an invalid scenario?

**Review by rjmccall — APPROVED** [review]
LGTM.  I should've pushed harder on that review, sorry

**Review by fhahn — APPROVED** [review]
Thanks for adding a test. LGTM as workaround

**Review by tkremenek — APPROVED** [review]

**Comment by aschwaighofer** [comment]
@swift-ci test

**Comment by aschwaighofer** [comment]
@swit-ci test

**Comment by aschwaighofer** [comment]
@swift-ci test

**Pull request merged**